src/__tests__/waitFor.test.tsx

```diff
@@ -79,21 +79,31 @@ test('waits for element with custom interval', async () => {
   expect(mockFn).toHaveBeenCalledTimes(2);
 });
 
-test.each([TimerMode.Legacy, TimerMode.Modern])(
-  'waits for element until it stops throwing using %s fake timers',
-  async (fakeTimerType) => {
-    jest.useFakeTimers(fakeTimerType);
-    const { getByText, queryByText } = render(<BananaContainer />);
+test('waits for element until it stops throwing using modern fake timers', async () => {
+  jest.useFakeTimers('modern');
```

Comment:
So the test works, it just needs to be using two separate `test` entries instead of one `test.each` entries.

Reply:
In my testing the test worked fine when `each` cause had only one of `modern` and `legacy` entries, but failed when there were both of them. I have a feeling that his is somehow connected to after test cleanup since splitting tests helped.

@thymikee maybe it make more sense to you as you know more about Jest internals.